scripts/cve_check.py

```diff
@@ -0,0 +1,642 @@
+                {
+                    "name": pkginfo["source"],
+                    "binary package name": pkginfo["bin_pkgs"],
+                    "version": str(pkginfo["version"]),
+                    "products": [
+                        {
+                            "product": product,
+                            "cvesInRecord": cvesInRecord,
+                        },
+                    ],
+                    "issue": issues,
+                },
+            ],
+        }
+
+        with open(filename, "w") as f:
+            json.dump(data, f, indent=4, sort_keys=False)
+
+    return sorted(filenames)
+
+def write_all_one_json(output_dir, image_name, json_filenames):
+    all_in_one_data = {
+        "version": "1",
+        "package": [],
+    }
+    
+    for filename in json_filenames:
+        with open(filename, "r") as f:
+            data = json.load(f)
+            all_in_one_data["package"].append(data["package"])
+
+    output_file = f"{output_dir}/{image_name}_cve.json"
+    with open(output_file, "w") as f:
+        json.dump(all_in_one_data, f, indent=4, sort_keys=False)
+
+def create_directory(target):
+    if not os.path.exists(target):
+        os.makedirs(target)
+
+def main(args):
+    nvd_cve.set_verbose_output(args.verbose_output)
+
+    bitbakeinfo = bitbake_runner.get_bitbake_information(args.image_name)
+    
+    cve_data_dl_dir = f"{bitbakeinfo['dl_dir']}/CVE"
+    create_directory(cve_data_dl_dir)
+
+    db_file = nvd_cve.update_nvd_db(cve_data_dl_dir, args.nvd_api_key)
+    if db_file is None:
+        print("Faied to fetch CVE database from NVD")
+        exit(1)
+
+    debian_cve_list = debian_cve.fetch_cve_data(cve_data_dl_dir)
+    if debian_cve_list is None:
+        print("Failed to fetch CVE data from Debian")
+        exit(1)
+
+    cip_kernel_sec_dir = kernel_cve.fetch_cip_kernel_sec(cve_data_dl_dir)
+    if cip_kernel_sec_dir is None:
+        print("Failed to fetch kernel-cip-sec")
+        exit(1)
+
+    installed_pkgs = debian_cve.parse_dpkg_status(bitbakeinfo["dpkg_status"])
+    cve_products = get_cve_products(args.extra_cve_product)
+    cve_ignore_list = get_cve_ignore(args.extra_cve_check_ignore)
+
+    uniq_installed_pkgs = create_unique_package(installed_pkgs)
+    linux_kernel_src_dir = os.path.abspath(bitbakeinfo["kernel_srcdir"])
+    cves = create_cves_info(db_file, debian_cve_list, uniq_installed_pkgs, installed_pkgs, args.debian_codename, cve_products, linux_kernel_src_dir, cip_kernel_sec_dir)
+
+    cves = update_ignored_cves_status(cves, cve_ignore_list)
+
+    output_base_dir = f"{bitbakeinfo['deploy_dir']}/cve/{bitbakeinfo['image_full_name']}"
+
+    create_directory(output_base_dir)
+
+    formats = args.output_format.split(",")
+    for fmt in formats:
+        if fmt == "text":
+            text_output_dir = f"{output_base_dir}/text"
+            create_directory(text_output_dir)
+            write_text(cves, text_output_dir, uniq_installed_pkgs)
```

Comment:
We need something like `write_all_one_text` here as well.

Reply:
Added to write all in one file by write_all_in_one_text().